internal/k8s/reconciler/gateway.go

```diff
@@ -166,10 +223,6 @@ func (g *K8sGateway) validatePodConditions(pod *corev1.Pod) {
 		return
 	}
 
-	if pod.Status.PodIP != "" {
```

Comment:
So, with this change and the `switch` above that only looks at service types, we'd no longer be populating the `addresses` field of the gateway status if no service type was defined. I'd personally prefer we just move this logic into a helper function and call it in the `if service == nil { ... }` block in the new `validateGatewayIP` function. Otherwise gateways without service-level exposure aren't going to carry information about how we can access them.